tests/test_qpe.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+from numpy.testing import assert_, assert_equal, assert_allclose
+import unittest
+from qutip import Qobj, sigmax, sigmay, sigmaz, identity, basis, tensor
+from qutip_qip.circuit import QubitCircuit
+from qutip_qip.operations import gate_sequence_product, ControlledGate
+
+from qutip_qip.algorithms.qpe import qpe, CustomGate, create_controlled_unitary
+
+class TestQPE(unittest.TestCase):
+    """
+    A test class for the Quantum Phase Estimation implementation
+    """
+    
+    def test_custom_gate(self):
+        """
+        Test if CustomGate correctly stores and returns the quantum object
+        """
+        U = Qobj([[0, 1], [1, 0]])  
+        
+        custom = CustomGate(targets=[0], U=U)
+        
+        qobj = custom.get_compact_qobj()
+        assert_((qobj - U).norm() < 1e-12)
+        
+        assert_equal(custom.targets, [0])
+        
+    def test_controlled_unitary(self):
+        """
+        Test if create_controlled_unitary creates a correct controlled gate
+        """
+        U = Qobj([[0, 1], [1, 0]])  
+        
+        controlled_u = create_controlled_unitary(controls=[0], targets=[1], U=U)
+        
+        assert_equal(controlled_u.controls, [0])
+        assert_equal(controlled_u.targets, [1])
+        assert_equal(controlled_u.control_value, 1)
+        
+        assert_(controlled_u.target_gate == CustomGate)
```

Comment:
target_gate is the same as the `CustomGate` class? Shouldn't it be a Gate initialised by `U`?

Reply:
I got this now